resources/lib/kodiUtilities.py

```diff
@@ -340,21 +340,52 @@ def getMovieDetailsFromKodi(libraryId, fields):
 
 
 def checkAndConfigureProxy():
-    proxyActive = kodiJsonRequest({'jsonrpc': '2.0', "method": "Settings.GetSettingValue", "params": {
-        "setting": "network.usehttpproxy"}, 'id': 1})['value']
-    proxyType = kodiJsonRequest({'jsonrpc': '2.0', "method": "Settings.GetSettingValue", "params": {
-        "setting": "network.httpproxytype"}, 'id': 1})['value']
-
-    if proxyActive and proxyType == 0:  # PROXY_HTTP
+    try:
         proxyURL = kodiJsonRequest({'jsonrpc': '2.0', "method": "Settings.GetSettingValue", "params": {
             "setting": "network.httpproxyserver"}, 'id': 1})['value']
+    except:
+        logger.debug("Failed to retrieve a proxyURL")
+
+    try:
         proxyPort = str(kodiJsonRequest({'jsonrpc': '2.0', "method": "Settings.GetSettingValue", "params": {
             "setting": "network.httpproxyport"}, 'id': 1})['value'])
+    except:
+        logger.debug("Failed to retrieve a proxyPort")
+
+    try:
         proxyUsername = kodiJsonRequest({'jsonrpc': '2.0', "method": "Settings.GetSettingValue", "params": {
             "setting": "network.httpproxyusername"}, 'id': 1})['value']
+    except:
+        logger.debug("Failed to retrieve a proxyUsername")
+
+    try:
         proxyPassword = kodiJsonRequest({'jsonrpc': '2.0', "method": "Settings.GetSettingValue", "params": {
             "setting": "network.httpproxypassword"}, 'id': 1})['value']
+    except:
+        logger.debug("Failed to retrieve a proxyPassword")
+
+    if not getSetting('proxy_uri'):
+        if proxyURL:
+            setSetting('proxy_uri', proxyURL)
+        if proxyPort:
+            setSetting('proxy_port', int(proxyPort))
+        if proxyUsername:
+            setSetting('proxy_user', str(proxyUsername))
+        if proxyPassword:
+            setSetting('proxy_passwd', proxyPassword)
```

Comment:
Not a fan of this. It's bad enough, kodi knows the password, but we shouldn't get it and save it to our settings. Even the proxyUrl and usernames might be sensitive in this context.

This also does it, when we don't even end up using it.

Reply:
Security vs user-friendliness :) I didn't think much of it as it was easy to extract and it was being sent-all over the place already anyway.
The goal was to have these fields pre-filled for the user to make things easier.

But I hear you, so i'll remove the prefilling. I agree security is more important here, and leaking this information should prevented.